### PR TITLE
Add wave-ice coupling to nuopc/cmeps driver

### PR DESCRIPTION
**Pull request by DeniseWorthen**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [x] Short (1 sentence) summary of your PR: 

Adds wave-ice coupling capability to nuopc/cmeps driver

- [x] Developer(s): 
    I believe the original changes were obtained from @lettie-roach during development of the new WW3 NUOPC/CMEPS cap. 
- [x] Suggest PR reviewers from list in the column to the right. @dabail10 @lettie-roach @apcraig 
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
    
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit wave-ice coupling is optional and has no impact unless enabled
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [x] Please provide any additional information or relevant details below: 

- The initialization of ``aicen`` and ``aicen_init`` in ice_state.F90 are required in order to run in debug mode when the intel compile option``-init=snan,arrays`` is used. 

- At the time this was committed to the NOAA-EMC fork, a simple comparison of ice thickness and concentration with and w/o wave-ice coupling enabled was made and the results were reasonable and in-line with expected behaviour (see figure in https://github.com/ufs-community/ufs-weather-model/pull/1381)

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
Seeing the changes in PR #775 related to FSD history variables, is the use of ``aicen_init`` for the calculation of floe_thickness and floe_diameter correct?

**Comment by dabail10** [comment]
This looks good. I will have to merge this into CESM and test it out. A few of these changes were already made in our cap.

**Comment by dabail10** [comment]
We might need a squash merge here. Lots of history.

File: cicecore/cicedynB/general/ice_state.F90 [hunk 1]
```diff
@@ -192,6 +192,8 @@ subroutine alloc_state
       n_trcr_strata = 0
       nt_strata = 0
       trcr_base = c0
+      aicen = c0
+      aicen_init = c0
```
Comment (dabail10) [inline cicecore/cicedynB/general/ice_state.F90:196]: For completeness, we should probably add vsnon/vsnon_init and vicen/vicen_init.

File: cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90 [hunk 2]
```diff
@@ -25,6 +25,7 @@ module ice_comp_nuopc
   use ice_calendar       , only : force_restart_now, write_ic
   use ice_calendar       , only : idate, mday, mmonth, myear, year_init
   use ice_calendar       , only : msec, dt, calendar, calendar_type, nextsw_cday, istep
+  use ice_calendar       , only : ice_calendar_noleap, ice_calendar_gregorian
```
Comment (dabail10) [inline cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90:28]: We had made this change in our cap already.

File: cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90 [hunk 3]
```diff
@@ -581,9 +579,11 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
...
     if (tfrz_option_driver  /= tfrz_option) then
-       write(errmsg,'(a)') trim(subname)//'error: tfrz_option from driver '//trim(tfrz_option_driver)//&
-            ' must be the same as tfrz_option from cice namelist '//trim(tfrz_option)
-       call abort_ice(trim(errmsg))
+       write(errmsg,'(a)') trim(subname)//'WARNING: tfrz_option from driver '//trim(tfrz_option_driver)//&
+            ' is overwriting tfrz_option from cice namelist '//trim(tfrz_option)
+       write(nu_diag,*) trim(errmsg)
+       call icepack_warnings_flush(nu_diag)
+       call icepack_init_parameters(tfrz_option_in=tfrz_option_driver)
     endif
```
Comment (dabail10) [inline cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90:587]: We also have this in our version.

**Comment by apcraig** [comment]
Default is to squash merge.  I think it's a good way to work on main.  Each PR looks like a single commit.  There are downsides, but minor.

**Comment by DeniseWorthen** [comment]
@dabail10 For testing wave-ice coupling, for completeness you should also take a look at this bug report (https://github.com/NOAA-EMC/WW3/issues/843). I have a fix going into the WW3 cap today or tomorrow, but I don't know when that change might make it to ESCOMP.

File: cicecore/cicedynB/general/ice_state.F90 (hunk 1 above)
Comment (DeniseWorthen) [reply cicecore/cicedynB/general/ice_state.F90:196]: I can add these initializations.

**Comment by eclare108213** [comment]
@DeniseWorthen 
The figures at https://github.com/ufs-community/ufs-weather-model/pull/1381) look great!  I see that the test runs used the floe size distribution.  Rather than comment there, let me ask a couple of questions here (where everyone knows me):
Were the waves only impacting the sea ice or were sea ice changes feeding back on the waves?
What is the spatial resolution?  Was that using C-grid in CICE?

**Comment by DeniseWorthen** [comment]
@eclare108213 The wave model sends the wave_elevation_spectrum to CICE6; CICE6 sends the floediam and ice thickness back to WW3. WW3 uses those variables, but I'm not well-versed enough in wave-ice coupling to say exactly how they interact w/ the wave field. 

The test case I ran used FV3 at C96 resolution and MOM6/CICE6 and WW3 all running on the 1-deg tripole grid. CICE6 was not using the C-grid.

**Comment by dabail10** [comment]
Sorry to take so long. I had to manually put the changes into my version of the CAP. I will issue a PR for my changes later. I did the aux_cice testsuite and this was bfb with cheyenne intel against my baselines. I will approve this and then merge into my code.

**Review by dabail10 — APPROVED** [review]

**Review by apcraig — APPROVED** [review]

**Comment by apcraig** [comment]
@dabail10 @DeniseWorthen, OK to merge, right?

**Comment by dabail10** [comment]
Good to go.

**Pull request merged**